source/MRMesh/MRVoxelsLoad.cpp

```diff
@@ -570,30 +565,88 @@ Expected<DicomVolume, std::string> loadDicomFolder( const std::filesystem::path&
     return res;
 }
 
-Expected<LoadDCMResult, std::string> loadDCMFolder( const std::filesystem::path& path,
-                                                     unsigned maxNumThreads, const ProgressCallback& cb )
+std::vector<Expected<DicomVolume, std::string>> loadDicomFolder( const std::filesystem::path& path,
+                                                        unsigned maxNumThreads, const ProgressCallback& cb )
 {
-    auto simple = loadDicomFolder( path, maxNumThreads, subprogress( cb, 0, 0.5f ) );
-    if ( !simple.has_value() )
-        return unexpected( std::move( simple.error() ) );
+    std::error_code ec;
+    if ( !std::filesystem::is_directory( path, ec ) )
+        return { unexpected( "loadDCMFolder: path is not directory" ) };
+
+    int filesNum = 0;
+    std::vector<std::filesystem::path> files;
+    for ( auto entry : Directory{ path, ec } )
+    {
+        if ( entry.is_regular_file( ec ) )
+            ++filesNum;
+    }
+
+    std::unordered_map<std::string, std::vector<std::filesystem::path>> seriesMap;
+    int fCounter = 0;
+    for ( auto entry : Directory{ path, ec } )
+    {
+        ++fCounter;
+        auto filePath = entry.path();
+        std::string uid;
+        if ( entry.is_regular_file( ec ) && isDICOMFile( filePath, uid ) )
+            seriesMap[uid].push_back( filePath );
+        if ( !reportProgress( cb, 0.3f * float( fCounter ) / float( filesNum ) ) )
+            return { unexpected( "Loading canceled" ) };
+    }
 
-    LoadDCMResult res;
-    res.vdbVolume = simpleVolumeToVdbVolume( simple->vol, subprogress( cb, 0.5f, 1.0f ) );
-    res.name = std::move( simple->name );
-    res.xf = std::move( simple->xf );
+    int seriesCounter = 0;
+    auto seriesNum = seriesMap.size();
+    std::vector<Expected<DicomVolume, std::string>> res;
+    for ( auto& [uid, series] : seriesMap )
+    {
+        res.push_back( loadSingleDicomFolder( series, maxNumThreads,
+            subprogress( cb,
+                0.3f + 0.7f * float( seriesCounter ) / float( seriesNum ),
+                0.3f + 0.7f * float( ++seriesCounter ) / float( seriesNum ) ) ) );
```

Comment:
Do not put `seriesCounter` and `++seriesCounter` in one line - it is a source of problems